### ISSUE TITLE
False positive "Unable to import 'requests.packages.urllib3.util.retry'"

### ISSUE THREAD

**Issue by yellowhat** [PyCQA/pylint#7203]
### Bug description

Hi,
consider the following:

```bash
$ python -c "from requests.packages.urllib3.util.retry import Retry"; echo $?
0
$ python -c "import requests.packages.urllib3.util.retry"; echo $?
0
```

But pylint is flagging `Unable to import 'requests.packages.urllib3.util.retry'`

Seems a false positive to me.

Thanks

### Command used

```shell
pylint --verbose --min-similarity-lines 8 --max-args 10 --max-locals 20 --max-line-length 200 $(find . -iname "*.py")
```

### Pylint output

```shell
No config file found, using default configuration
************* Module config
config.py:8:0: E0401: Unable to import 'requests.packages.urllib3.util.retry' (import-error)
```

### Expected behavior

Succeed

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.10.5 (main, Jul 19 2022, 02:32:29) [GCC 11.2.1 20220219]
```

### Additional dependencies

podman run -it --rm -v "$PWD:/data" -w "/data" python:alpine sh

**Comment by clavedeluna** [comment]
You need to have `requests` installed to correctly reproduce this. After running `pip install requests` the following example will reproduce it:

```
# test.py
import requests
from requests.packages.urllib3.util.retry import Retry
import requests.packages.urllib3.util.retry
```

```
************* Module test
test.py:2:0: E0401: Unable to import 'requests.packages.urllib3.util.retry' (import-error)
test.py:3:0: E0401: Unable to import 'requests.packages.urllib3.util.retry' (import-error)
test.py:1:0: W0611: Unused import requests (unused-import)
```

since only an unused-import and not an import-error is given for the first line, we know the import-errors for  lines 2-3 are a false positive.

**Comment by mbyrnepr2** [comment]
Closing as a duplicate of https://github.com/PyCQA/pylint/issues/4584. The root cause appears to be the way `urllib3` is [imported directly](https://github.com/psf/requests/blob/main/requests/packages.py#L15) using the `__import__` function.
